packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -68,8 +68,15 @@ function targetGetter(this: Event): EventTarget | null {
     // and when an event has been added to Window
     if (!(originalCurrentTarget instanceof Node)) {
         // TODO: issue #1511 - Special escape hatch to support legacy behavior. Should be fixed.
-        // If the event's target is being accessed async and originalTarget is not a keyed element, do not retarget
-        if (isNull(originalCurrentTarget) && isUndefined(getNodeOwnerKey(originalTarget as Node))) {
+        // If the originalTarget is not a keyed element and if any of the following conditions are true, do not retarget
```

Comment:
I thought this was for 222 instead of master... also, this block seems to be for the cases where currentTarget is null or window...

Reply:
Good catch. Fixing it and adding a test.